applications/asset_tracker/src/cloud_codec/cloud_codec.c

```diff
@@ -283,7 +283,26 @@ int cloud_encode_digital_twin_data(const struct cloud_channel_data *channel,
 		return -ENOMEM;
 	}
 
-	ret = json_add_obj(reported_obj, channel_type_str[channel->type],
+	/* Workaround for deleting "DEVICE" objects (with uppercase key) if
+	 * it already exists in the digital twin.
+	 * The cloud implementations expect the key to be lowercase "device",
+	 * but that would duplicate the information and needlessly increase
+	 * the size of the digital twin document if the "DEVICE" is not
+	 * deleted at the same time.
+	 */
+	if (channel->type == CLOUD_CHANNEL_DEVICE_INFO) {
+		cJSON *dummy_obj = cJSON_CreateNull();
+		if (dummy_obj == NULL) {
+			cJSON_Delete(reported_obj);
+			/* Dummy creation failed, but we'll let it do so
+			 * silently as it's not a functionally critical error.
+			 */
+		} else {
+			ret += json_add_obj(reported_obj, "DEVICE", dummy_obj);
+		}
+	}
+
+	ret += json_add_obj(reported_obj, channel_type_str[channel->type],
```

Comment:
We could've removed the `reported_obj` just above, is it safe to use it here unconditionally?

Reply:
Oops, I'm dleting the wrong object... Good catch. It should be the dummy that's deleted. Actually, no need to delete it I guess, if it was never allocated and returned `NULL`

Reply:
Yes, I agree there's no need to delete it if it failed during allocation.

Reply:
Removed the deletion and rebased on to master.